internal/pkg/manifest/storage.go

```diff
@@ -69,50 +71,54 @@ func (e *EFSConfigOrID) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 	if !e.Config.IsEmpty() {
 		// Unmarshaled successfully to e.Config, unset e.ID, and return.
-		e.ID = ""
+		e.Enabled = nil
 		return nil
 	}
 
-	if err := unmarshal(&e.ID); err != nil {
-		return errUnmarshalBuildOpts
+	if err := unmarshal(&e.Enabled); err != nil {
+		return errUnmarshalEFSOpts
 	}
 	return nil
 }
 
-// UseManagedFS returns true if the user has specified "copilot" or "managed" as a FSID; false otherwise.
-func (e *EFSConfigOrID) UseManagedFS() bool {
-	if contains(managedFSIDKeys, e.ID) {
-		return true
+// UseManagedFS returns true if the user has specified EFS as a bool, or has only specified UID and GID.
+func (e *EFSConfigOrBool) UseManagedFS() bool {
+	// Respect explicitly enabled or disabled value first.
+	if e.Enabled != nil {
+		return aws.BoolValue(e.Enabled)
 	}
-	if contains(managedFSIDKeys, aws.StringValue(e.Config.FileSystemID)) {
+	// Check whether we're implicitly enabling managed EFS via UID/GID.
+	if !e.Config.EmptyUIDConfig() {
 		return true
 	}
+
 	return false
 }
 
-// FSID returns the correct value of the EFS filesystem ID. If the ID is set improperly (via a bad merge
-// of environment overrides), it throws an error.
-func (e *EFSConfigOrID) FSID() *string {
-	fromID := e.ID
-	// If config is empty, use ID; otherwise ignore string and use struct.
-	if e.Config.IsEmpty() {
-		return aws.String(fromID)
+func (e *EFSVolumeConfiguration) EmptyBYOConfig() bool {
+	return e.FileSystemID == nil && e.AuthConfig == nil && e.RootDirectory == nil
+}
+
+func (e *EFSVolumeConfiguration) EmptyUIDConfig() bool {
+	return e.UID == nil && e.GID == nil
+}
+
+func (e *EFSConfigOrBool) EmptyVolume() bool {
+	// Respect Bool value first: return true if Enabled is false; false if true.
+	if e.Enabled != nil {
+		return !aws.BoolValue(e.Enabled)
```

Comment:
nit:  ⬇️ might be easier to see the intention?
```suggestion
		return aws.BoolValue(e.Enabled) == false
```